os/board/rtl8721csm/src/component/common/api/wifi/wifi_conf.c

```diff
@@ -1272,6 +1272,9 @@ _WEAK void wifi_set_mib(void)
 	wext_set_adaptivity(RTW_ADAPTIVITY_DISABLE);
 	//trp tis
 	wext_set_trp_tis(DISABLE);
+
+	//send deauth before auth to disconnect from AP to resolve AP compatibility issue

```

Comment:
is it the case of disconnect and connect to same AP here?

Reply:
yes, the compatibility issue arises in some AP when it tries to connect to same AP before disconnecting.